src/hooks/data/useVaultApy.ts

```diff
@@ -76,9 +161,167 @@ export function useVaultTotalApr({
   poolId,
   type,
 }: VaultDynamicProps) {
+  const isCurve = useIsCurve(type)
+  const isToken = useIsTokenCompounder(type)
   // Preferred: API request
   const apiQuery = useApiVaultDynamic({ poolId, type })
-  // TODO: Fallbacks?
+
+  // CURVE
+  const vaultAprFallback = useQuery([_address, "vaultAprFallback"], {
+    queryFn: async () => await getVaultAprFallback(_address),
+    retry: false,
+    enabled:
+      (apiQuery.isError || apiQuery.data === null) && !isToken && isCurve,
+  })
+  // END OF CURVE
+
+  // BALANCER
+  const auraQuery = useQueries({
```

Comment:
This file is a bit overwhelming. It might be more readable/manageable if you broke these fallback queries + the parsing involved for each into separate hooks. Something like this:


```
hooks/data
hooks/data/aprFallbacks
hooks/data/aprFallbacks/useCurveVaultGraphToalApr
hooks/data/aprFallbacks/useBalancerVaultGraphTotalApr
hooks/data/aprFallbacks/useTokenVaultGraphTotalApr
...
```

```ts
const isCurveFallbackEnabled = (apiQuery.isError || apiQuery.data === null) && isCurve && !isToken
const isBalancerFallbackEnabled = (apiQuery.isError || apiQuery.data === null) &&  !isCurve && !isToken
//...

const curveVaultGraphTotalApr = useCurveVaultGraphTotalApr({ enabled: isCurveFallbackEnabled })
const balancerVaultGraphTotalApr = useBalancerVaultGraphTotalApr({ enabled: isBalancerFallbackEnabled }}
// ...

if (isCurveFallbackEnabled) {
  return curveVaultGraphTotalApr
}
if (isBalancerFallbackEnabled) {
  return balancerVaultGraphTotalApr
}
// ...
```

Reply:
yes, good idea